Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/TempSpawnableAssetsCache.cpp

```diff
@@ -0,0 +1,225 @@
+#include <AzToolsFramework/Prefab/PrefabLoader.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+
+namespace AzToolsFramework::Prefab::PrefabConversionUtils
+{
+    TempSpawnableAssetsCache::~TempSpawnableAssetsCache()
+    {
+        Deactivate();
+    }
+
+    bool TempSpawnableAssetsCache::Activate(AZStd::string_view stackProfile)
+    {
+        AZ_Assert(!IsActivated(), "TempSpawnableAssetsCache - TempSpawnableAssetsCache is currently activated");
+
+        m_prefabSystemComponentInterface = AZ::Interface<PrefabSystemComponentInterface>::Get();
+        AZ_Assert(m_prefabSystemComponentInterface, "TempSpawnableAssetsCache - Could not retrieve instance of PrefabSystemComponentInterface");
+
+        m_loaderInterface = AZ::Interface<Prefab::PrefabLoaderInterface>::Get();
+        AZ_Assert(m_loaderInterface, "TempSpawnableAssetsCache - Could not retrieve instance of PrefabLoaderInterface");
+
+        m_stockProfile = stackProfile;
+        return m_converter.LoadStackProfile(m_stockProfile);
+    }
+
+    void TempSpawnableAssetsCache::Deactivate()
+    {
+        ClearAllTempSpawnableRelatedAssets();
+        m_stockProfile = "";
+        m_loaderInterface = nullptr;
+        m_prefabSystemComponentInterface = nullptr;
+    }
+
+    bool TempSpawnableAssetsCache::IsActivated() const
+    {
+        return m_converter.IsLoaded();
+    }
+
+    AZStd::string_view TempSpawnableAssetsCache::GetStockProfile() const
+    {
+        return m_stockProfile;
+    }
+
+    TempSpawnableAssetsCache::CreateSpawnableResult TempSpawnableAssetsCache::CreateTempSpawnableAsset(
+        AzToolsFramework::Prefab::TemplateId templateId, AZStd::string_view spawnableName, bool loadCreatedAssets)
+    {
+        if (!IsActivated())
+        {
+            return AZ::Failure(AZStd::string::format("Failed to create a prefab processing stack from key '%.*s'.", AZ_STRING_ARG(m_stockProfile)));
+        }
+
+        TemplateReference templateReference = m_prefabSystemComponentInterface->FindTemplate(templateId);
+        if (!templateReference.has_value())
+        {
+            return AZ::Failure(AZStd::string::format("Could not get Template DOM given Template's id %llu .", templateId));
+        }
+
+        // Use a random uuid as this is only a temporary source.
+        PrefabConversionUtils::PrefabProcessorContext context(AZ::Uuid::CreateRandom());
+        PrefabDom copy;
+        copy.CopyFrom(templateReference->get().GetPrefabDom(), copy.GetAllocator(), false);
+        context.AddPrefab(spawnableName, AZStd::move(copy));
+        m_converter.ProcessPrefab(context);
+
+        if (!context.HasCompletedSuccessfully() || context.GetProcessedObjects().empty())
+        {
+            return AZ::Failure(AZStd::string(
+                "Failed to convert the prefab into assets."
+                "Please confirm that the 'TempSpawnables' prefab processor stack is capable of producing a usable product asset."));
```

Comment:
Should "TempSpawnables' be the name of the provided configuration stored in `m_stockProfile`?